ui-tests/src/main/java/org/openmrs/reference/page/AppointmentBlocksPage.java

```diff
@@ -10,105 +10,222 @@
+	
+	private static final By SERVICE_BLOCK = By.className("fc-event-inner");
+	
+	private static final By DAY_BLOCK = By.xpath("//table[@class='fc-border-separate']/tbody/tr/td/div");
+	
+	private static final By REMOVE_APPOINTMENT = By
+	        .xpath("//div[@id='appointment-block-form']/selectmultipleappointmenttypes/div/div/div/div/i");
+	
+	private static final By CANCEL = By.className("cancel");
+	
+	public void goToAppointmentBlock() {
+		clickOn(APPOINTMENT_SCHEDULING);
+		clickOn(MANAGE_PROVIDER_SCHEDULES);
+	}
+	
+	/**
+	 * Removes an appointment. Before calling this method is necessary to click on the day (User
+	 * should click on the day button on the left)
+	 * 
+	 * @see #clickOnDay()
+	 */
+	public void removeAppointment() {
+		clickOn(REMOVE_APPOINTMENT);
+	}
+	
+	public void selectLocation(String location) {
+		// a wait might be needed if there are many elements to load in the dropdown menu
+		waiter.until(ExpectedConditions.elementToBeClickable(LOCATION));
+		selectFrom(LOCATION, location);
+		clickOn(LOCATION);
+	}
+	
+	public void selectLocationBlock(String locblock) {
+		waitForElement(LOCATION_IN_BLOCK);
+		selectFrom(LOCATION_IN_BLOCK, locblock);
+		clickOn(LOCATION_IN_BLOCK);
+	}
+	
+	public void enterService(String service) {
+		boolean flag = false;
+		while (!flag) {
+			try {
+				findElement(SERVICE).clear();
+				setTextToFieldNoEnter(SERVICE, service);
+				waitForElement(SERVICE_DROPDOWN);
+				clickOn(SERVICE_DROPDOWN);
+				flag = true;
+			}
+			catch (Exception e) {
+				flag = false;
+			}
+		}
+		
+	}
+	
+	public void enterProvider(String provider) {
+		setTextToFieldNoEnter(PROVIDER, provider);
+	}
+	
+	public void clickOnCurrentDay() throws InterruptedException {
+		clickOn(CURRENT_DAY);
+	}
+	
+	public void enterStartTime(String start) {
+		setTextToFieldNoEnter(START_TIME, start);
+	}
+	
+	/**
+	 * Scrolls the page up This method has been created since the delete button is hidden by the
+	 * tooltip. An alternative solution to this method is to create a method that closes the tooltip
+	 * such as: private static final By CLOSE_TOOLTIP =
+	 * By.cssSelector("span.ui-icon.ui-icon-close"); public void clickOnCloseTooltip(){
+	 * clickOn(CLOSE_TOOLTIP); } Unfortunately this method did not prove to be always reliable
+	 * (sometime Selenium fails to locate span.ui-icon.ui-icon-close)
+	 */
+	public void clickOnleft() {
+		executeScript("scroll(300, 0)");
+	}
+	
+	/**
+	 * Clicks on the current day and open the "editing page". This method should be invoked when the
+	 * page shows only a block with the day
+	 */
+	public void clickOnDay() {
+		clickOn(DAY);
+		clickOn(DAY_CLASS);
+	}
+	
+	/**
+	 * Clicks on the Cancel button available in the "editing page"
+	 **/
+	public void clickOnCancel() {
+		clickOn(CANCEL);
+	}
+	
+	public void clickOnSave() {
+		clickOn(SAVE);
+		try {
+			waitForElementToBeHidden(SAVE);
+		}
+		catch (Exception e) {
+			
```

Comment:
It was a code already existing when I started to work on the class, I'll clean it.